Documentation/CMake.md

```diff
@@ -59,21 +59,36 @@ endif()
 ## Add an entry point
 
 You must include a source file in your test executable target with a
-`@main` entry point. The following example uses the SwiftPM entry point:
+`@main` entry point. The example main below requires the experimental
+`Extern` feature. The declaration of `swt_abiv0_getEntryPoint` could
+also be written in a C header file with its own `module.modulemap`.
 
 ```swift
-import Testing
+import Foundation
```

Comment:
Shouldn't need this import for the barebones implementation below. Only needed if you want to use `JSONEncoder`/`JSONDecoder`.

Reply:
Is it not needed for `exit(EXIT_SUCCESS)`? Or is there a non-foundation way to do the standard Unix process exit dance.

Reply:
`exit()` and `EXIT_SUCCESS` are defined in the system's C module, which may be `Darwin`, `Glibc`, `ucrt`… uh… `Musl`?

Foundation reexports the C module for the current system. It seems wrong to me (but I can be convinced otherwise) to tell developers to include a full Foundation dependency just for `exit()`.

Depending on the dev's specific requirements, they really just need the process to terminate with zero (the default!) or non-zero. If they're using Swift Argument Parser, it has built-in functionality for this. If they're using _just_ the Swift stdlib and nothing else, they could call `fatalError()` on failure. I don't know what's the best choice for this example code.

Reply:
Eh.. I'm not sure that I know what the best choice is either. When a test binary is executed by ctest, as in my use case, returning 0 or 1 from main() is important for the test harness to understand whether the test passed or failed. Anything more than that on the ctest side is up to whatever @etcwilde comes up with for https://gitlab.kitware.com/cmake/cmake/-/issues/26452

I feel like adding a 
```
#if canImport(GlibC)
#import GlibC
#elseif canImport(YourCLibModule)
#import YourCLibModule
#endif
```
chain to the top of the example would get in the way of what it's actually trying to show that you need :shrug: 

I'll change it to whatever anyone feels strongly about though, as in theory more comprehensive support for the CMake use case could come soon :tm: to make this example obsolete-ish.

Reply:
It may be sufficient to just show `exit(EXIT_SUCCESS)` without importing anything. Developers who are going down this path are presumably skilled enough to figure out how to make it compile…? 😇

Reply:
Sure... "compilable example code left as an exercise for the reader" is definitely one approach. Not my first choice, but it's not my library :). I've pushed up that change then.

Reply:
Again, anybody who's digging down into the CMake scripts is on their own and I have to assume has some esoteric environment—we can't even guarantee Foundation is present! (We _can_ guarantee libc is present because we won't build without it, at least.)